### PR TITLE
WHO/LCI panharmonicons and tweaks

### PR DESCRIPTION
**Pull request by Northmoc**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/clara_oswald.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Name:Clara Oswald
+ManaCost:6
+Types:Legendary Creature Human Advisor
+PT:2/6
+S:Mode$ Continuous | EffectZone$ All | CharacteristicDefining$ True | Affected$ Card.Self | SetColor$ ChosenColor | Description$ Impossible Girl — If CARDNAME is your commander, choose a color before the game begins. CARDNAME is the chosen color.
+S:Mode$ Panharmonicon | ValidCard$ Card.IsDoctor | Description$ If a triggered ability of a Doctor you control triggers, that ability triggers an additional time.
```
Comment (Glorax) [inline forge-gui/res/cardsfolder/upcoming/clara_oswald.txt:6]: Needs to be a Doctor you control, not just any Doctor.

File: forge-game/src/main/java/forge/game/card/CardProperty.java [hunk 2]
```diff
@@ -178,6 +178,10 @@ public static boolean cardHasProperty(Card card, String property, Player sourceC
             if (!card.isRingBearer()) {
                 return false;
             }
+        } else if (property.equals("IsDoctor")) {
+            if (!card.getRules().isDoctor()) {
+                return false;
+            }
```
Comment (Glorax) [inline forge-game/src/main/java/forge/game/card/CardProperty.java:184]: Is this actually necessary? Harmonic Prodigy has "ValidCard$ Shaman.YouCtrl,Wizard.Other+YouCtrl", so wouldn't "ValidCard$ Doctor.YouCtrl" work fine?

File: forge-gui/res/cardsfolder/upcoming/roaming_throne.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Name:Roaming Throne
+ManaCost:4
+Types:Artifact Creature Golem
+PT:4/4
+K:Ward:2
+K:ETBReplacement:Other:ChooseCT
+SVar:ChooseCT:DB$ ChooseType | Type$ Creature | AILogic$ MostProminentInComputerDeck | SpellDescription$ As CARDNAME enters the battlefield, choose a creature type.
+S:Mode$ Continuous | Affected$ Card.Self | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+S:Mode$ Panharmonicon | ValidCard$ Card.ChosenType | Description$ If a triggered ability of another creature you control of the chosen type triggers, it triggers an additional time.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/roaming_throne.txt:9]: wrong Valid

File: forge-game/src/main/java/forge/game/card/CardProperty.java (hunk 2 above)
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/card/CardProperty.java:184]: checking rules is very wrong, let me know if I should explain further

File: forge-gui/res/cardsfolder/upcoming/roaming_throne.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Name:Roaming Throne
+ManaCost:4
+Types:Artifact Creature Golem
+PT:4/4
+K:Ward:2
+K:ETBReplacement:Other:ChooseCT
+SVar:ChooseCT:DB$ ChooseType | Type$ Creature | AILogic$ MostProminentInComputerDeck | SpellDescription$ As CARDNAME enters the battlefield, choose a creature type.
+S:Mode$ Continuous | Affected$ Card.Self | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+S:Mode$ Panharmonicon | ValidCard$ Creature.Other+YouCtrl+ChosenType | Description$ If a triggered ability of another creature you control of the chosen type triggers, it triggers an additional time.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/roaming_throne.txt:9]: It says "creature you control", so it needs inZoneBattlefield check

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/roaming_throne.txt:9]: should be handled?
image

**Pull request merged**